src/checker.ts

```diff
@@ -0,0 +1,227 @@
+import * as ts from "typescript";
+import { AppsyncResolver } from "./appsync";
+import { EventBus, EventBusRule } from "./event-bridge";
+import { EventBusTransform } from "./event-bridge/transform";
+import { ExpressStepFunction, StepFunction } from "./step-function";
+
+/**
+ * Various types that could be in a call argument position of a function parameter.
+ */
+export type TsFunctionParameter =
+  | ts.FunctionExpression
+  | ts.ArrowFunction
+  | ts.Identifier
+  | ts.PropertyAccessExpression
+  | ts.ElementAccessExpression
+  | ts.CallExpression;
+
+export type EventBusRuleInterface = ts.NewExpression & {
+  arguments: [any, any, any, TsFunctionParameter];
+};
+
+export type EventBusTransformInterface = ts.NewExpression & {
+  arguments: [TsFunctionParameter, any];
+};
+
+export type EventBusWhenInterface = ts.CallExpression & {
+  arguments: [any, any, TsFunctionParameter];
+};
+
+export type EventBusMapInterface = ts.CallExpression & {
+  arguments: [TsFunctionParameter];
+};
+
+export interface FunctionlessChecker extends ts.TypeChecker {
+  isAppsyncResolver(node: ts.Node): node is ts.NewExpression & {
+    arguments: [TsFunctionParameter, ...ts.Expression[]];
+  };
+  isEventBusRuleMapFunction(node: ts.Node): node is EventBusMapInterface;
+  isEventBusWhenFunction(node: ts.Node): node is EventBusWhenInterface;
+  isFunctionlessType(type: ts.Type | undefined, kind: string): boolean;
+  isNewEventBusRule(node: ts.Node): node is EventBusRuleInterface;
+  isNewEventBusTransform(node: ts.Node): node is EventBusTransformInterface;
+  isReflectFunction(node: ts.Node): node is ts.CallExpression & {
+    arguments: [TsFunctionParameter, ...ts.Expression[]];
+  };
+  isStepFunction(node: ts.Node): node is ts.NewExpression & {
+    arguments: [TsFunctionParameter, ...ts.Expression[]];
+  };
+}
+
+export function makeFunctionlessChecker(
+  checker: ts.TypeChecker
+): FunctionlessChecker {
+  return {
+    ...checker,
+    isAppsyncResolver,
+    isEventBusRuleMapFunction,
+    isEventBusWhenFunction,
+    isFunctionlessType,
+    isNewEventBusRule,
+    isNewEventBusTransform,
+    isReflectFunction,
+    isStepFunction,
+  };
```

Comment:
Any way to avoid maintaining the signatures in 2 places?

Reply:
How about this? And then we just leave the type as inferred. 

```
ReturnType<typeof makeFunctionlessChecker>
```